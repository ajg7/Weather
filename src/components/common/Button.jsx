import React from "react";
import PropTypes from "prop-types";

const Button = props => {
	const { buttonText, clickFunc, value } = props;
	return (
		<div>
			<button onClick={clickFunc} value={value}>
				{buttonText}
			</button>
		</div>
	);
};

Button.propTypes = {
	buttonText: PropTypes.string,
	clickFunc: PropTypes.func,
	value: PropTypes.string,
};

export default Button;
