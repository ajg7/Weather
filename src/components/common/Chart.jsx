import React from "react";

const Chart = () => {
	return (
		<div>
			<canvas></canvas>
		</div>
	);
};

export default Chart;
