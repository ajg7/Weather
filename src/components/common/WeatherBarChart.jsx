import React, { useRef } from "react";
import PropTypes from "prop-types";
import { getDates } from "../../utils/getDates";
import { useBarGraph } from "../../hooks";

const WeatherBarChart = props => {
	const { tempData } = props;
	const chartRef = useRef();
	const dates = getDates();
	useBarGraph(chartRef, tempData, ["Average Highs", "Average Lows"], dates);
	return (
		<div>
			<canvas ref={chartRef} width="800" height="400"></canvas>
		</div>
	);
};

WeatherBarChart.propTypes = {
	tempData: PropTypes.array,
};

export default WeatherBarChart;
