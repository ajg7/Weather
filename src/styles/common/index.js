import { StyledButton } from "./StyledButton";
import { StyledTitle } from "./StyledTitle";
import { StyledDescriptor } from "./StyledDescriptor";

export { StyledButton, StyledTitle, StyledDescriptor };