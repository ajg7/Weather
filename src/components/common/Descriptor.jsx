import React from "react";
import PropTypes from "prop-types";
import { StyledDescriptor } from "../../styles/common";

const Descriptor = props => {
	const { description, headingNumber } = props;
	return <StyledDescriptor className={"descriptor"}>{headingNumber === 2 ? <h2>{description}</h2> : <h3>{description}</h3>}</StyledDescriptor>;
};

Descriptor.propTypes = {
	description: PropTypes.string,
	headingNumber: PropTypes.number,
};

export default Descriptor;
