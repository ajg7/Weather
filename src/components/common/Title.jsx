import React from "react";
import PropTypes from "prop-types";
import { StyledTitle } from "../../styles/common";

const Title = props => {
	const { title } = props;
	return (
		<StyledTitle className={"title"}>
			<h1>{title}</h1>
		</StyledTitle>
	);
};

Title.propTypes = {
	title: PropTypes.string,
};

export default Title;
