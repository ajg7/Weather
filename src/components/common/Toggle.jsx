import React from "react";
import PropTypes from "prop-types";
import { StyledToggle } from "../../styles/common";

const Toggle = props => {
	const { toggleLabel, changeFunc, classNameText } = props;
	return (
		<StyledToggle>
			<label>{toggleLabel}</label>
			<input
				id={"toggle-input"}
				className={classNameText}
				type={"checkbox"}
				name={"toggleSwitch"}
				onChange={changeFunc}
			/>
		</StyledToggle>
	);
};

Toggle.propTypes = {
	toggleLabel: PropTypes.string,
	changeFunc: PropTypes.func,
	classNameText: PropTypes.string,
};

export default Toggle;
