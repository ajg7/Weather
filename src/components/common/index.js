import Descriptor from "./Descriptor";
import Title from "./Title";

export { Descriptor, Title };
