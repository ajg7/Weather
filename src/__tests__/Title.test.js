import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Title } from "../components/common";

configure({ adapter: new Adapter() });
const title = "Weather Test Title";
const wrapped = shallow(<Title title={title} />);

describe("Title", () => {
	test("should render the Title Component", () => {
		expect(wrapped).toMatchSnapshot();
	});

	test("renders the Title Text", () => {
		expect(wrapped.find("h1").text()).toEqual(title);
	});
});
