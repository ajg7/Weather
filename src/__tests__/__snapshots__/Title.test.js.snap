// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title should render the Title Component 1`] = `
<styled.div
  className="title"
>
  <h1>
    Weather Test Title
  </h1>
</styled.div>
`;
